### ISSUE TITLE
Heuristics for adding dynamic bones to avatar during avatar creation

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#699]
Adding heuristics for detection of hair, tails, etc. as a setting in the avatar creator could greatly speed up avatar setup, and enhance the visual quality of avatar.

**Comment by TehTurk** [comment]
While this isn't bad I don't rec this be implemented because as the proper process for disabling dynamic bones won't be know can break certain armatures.(if they moving they get stuck in their og position) and while this could be tweaked This also can result in some rigs being very unoptimized with 300like hair bone chains I've found with some users. Letting people customize their avatars is a good thing. Letting them figure and know is good as well.

**Comment by Frooxius** [comment]
This might be kinda tricky, because heuristics like this is hard to get right, so it does what the user expects. With such wide variety of rigs and naming schemes, it's hard to tell what and how exactly to setup the bones and it can easily end up messing up the rig.

Maybe it could be added at some point, but this is something that's better to just have the user setup manually.

**Comment by shiftyscales** [comment]
It probably would be tricky to implement in that specific form- it could also be resolved by having an easier/simpler interface or mechanism through which bone chains can be added/removed, as it will be a common feature a lot of users will want to incorporate into avatars they import.

Although that could also possibly be taken care of via user-generated content once component access is implemented, and doesn't necessarily need to be resolved in code.

**Comment by Frooxius** [comment]
Yeah a tool could potentially be made. E.g. something that visualizes the avatar rig, you click on a bone and make the whole chain into a dynamic bone and and setup a few defaults.

**Comment by dfgHiatus** [comment]
I would like to renew interest in this feature. Whenever I import (or reimport) an avatar, I find it tedious to component clone my dynamic bones onto many slots.

I imagine an installer (or tool) would do some string matching (breasts, tail, ears, etc) given an avatar's slot and apply/setup dynamic bones with given settings (stiffness, inertia, etc.). Imagine an asset optimization wizard but for dynamic bones. 

Personally,  I use the same settings for all my bones.
